### PR TITLE
web query functions

### PR DESCRIPTION
**Pull request by raipc**

### CODE REVIEW

File: rule-engine/functions-web-query.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+## Overview
+
+The functions in this section perform an HTTP request to external web service 
+and return the result in the object with following fields:
+
+**Field**    | **Type** | **Description**
+-------------|----------|----------------
+content      | string   | Response body
+headers      | map      | Map of HTTP headers. Header values with the same name are separated by a comma.
+contentType  | string   | Value of Content-Type header
```
Comment (rodionos) [inline rule-engine/functions-web-query.md]: - Rename responseCode to status.
- contentType is just one of headers, I would not expose (or at least document it)
- I'd hide reason phrase (or not document) to keep it simple.

File: rule-engine/functions-web-query.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+### `webNotify`
+
+```javascript
+  webNotify(string c, map p) response
+```
+
+Perform a request using an existing web configuration `c` passing to it the parameters `p`
+
+Example:
```
Comment (rodionos) [inline rule-engine/functions-web-query.md]: Does the name have to be case-sensitive?

File: rule-engine/functions-web-query.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+```javascript
+  queryUrl(string u) response
+```
+
+Perform a GET request to URL `u` using a temporary custom configuration
+
+Example:
+
+```javascript
+  queryUrl("https://api.telegram.org/bot1234567/getMe")
```
Comment (rodionos) [inline rule-engine/functions-web-query.md]: ipinfo.io/8.8.8.8 would be a better example, it does not require any preparation (create bot)

File: rule-engine/functions-web-query.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+```javascript
+  queryUrl(map c, map p) response
+```
+
+Perform a request using a temporary custom configuration 
+built using web configuration parameters `c`. 
+Request body is built from request parameters `p` according to provided content type.
+Allowed configuration parameters: "url", "method", "contentType"
+
```
Comment (rodionos) [inline rule-engine/functions-web-query.md]: Do we want to introduce some defaults to make it less verbose?
If posting json is a common use case, we could POST and application/json as default values.

File: rule-engine/functions-web-query.md (hunk 2 above)
Comment (raipc) [reply rule-engine/functions-web-query.md]: Yes, it has

File: rule-engine/functions-web-query.md (hunk 4 above)
Comment (raipc) [reply rule-engine/functions-web-query.md]: Default are the same as in Custom configuration: POST method and form/urlencoded content type

File: rule-engine/functions-web-query.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+* The web notification must be listed on the **Alerts > Web Notifications** page.
+* The name of the web notification `c` is case-sensitive.
+* The web notification must be enabled.
+
+The parameter map `p` is used to substitute placeholders in the given web notification.
+
+Example:
+
+```javascript
+  webNotify("slack-devops", ["text": "Alert"])
```
Comment (rodionos) [inline rule-engine/functions-web-query.md]: Add more information on what happens with `p` parameters. Do they replace placeholders in json? What if `p` is not empty and the JSON is empty or the JSON doesn't contain some of the keys in the `p` map?

File: rule-engine/functions-text.md [hunk 6]
```diff
@@ -424,3 +427,59 @@ Removes leading and trailing non-printable characters.
...
+
+```javascript
+  flattenJson(string j) map
+```
+
+Converts a JSON-object string into map. If element is nested, its key will be a 
+concatenation of all parent elements' names with own name, each separated by dot.
+
+```javascript
+  flattenJson(message)
```
Comment (rodionos) [inline rule-engine/functions-text.md]: An example showing how we obtained the json with `queryUrl` would be useful here.

Comment (rodionos) [reply rule-engine/functions-text.md]: The example should show an input json, preferably with nested objects, and the produced map.

**Comment by kghamilton89** [comment]
I've reviewed this PR, everything looks good.

**Pull request merged**